relay-server/src/actors/events.rs

```diff
@@ -1387,9 +1396,112 @@ impl Handler<QueueEnvelope> for EventManager {
     }
 }
 
+/// Takes an envelope and potentially removes the transaction item from it if that
+/// transaction item should be sampled out according to the dynamic sampling configuration
+/// and the trace context.
+fn sample_transaction_internal<'a>(
```

Comment:
This function looks like it can be a function on `ProjectState`.

Reply:
Sure, but that would be a bit strange, its primary operating structure is the Envelope ( which it might change). 
More to the point it is an implementation detail of "transaction sampling" so I would prefer to leave it as a private function inside `relay_server::utils::dynamic_sampling` (just moved there from `events.rs` together with `sample_transaction`).